Sources/App/Controllers/PackageController+routes.swift

```diff
@@ -103,20 +112,65 @@ struct PackageController {
 
         switch fragment {
             case .documentation:
-                // Try and parse the page and add our header, but fall back to the unprocessed page if it fails.
-                guard let queryResult = try await Joined3<Package, Repository, Version>
-                    .query(on: req.db, owner: owner, repository: repository, version: .defaultBranch)
+                let queryResult = try await Joined3<Version, Package, Repository>
+                    .query(on: req.db,
+                           join: \Version.$package.$id == \Package.$id, method: .inner,
+                           join: \Package.$id == \Repository.$package.$id, method: .inner)
+                    .filter(Repository.self, \.$owner == owner) //TODO: Lowercase
+                    .filter(Repository.self, \.$name == repository)
+                    .filter(\Version.$docArchives != nil)
+                    .field(Version.self, \.$reference)
+                    .field(Version.self, \.$latest)
                     .field(Version.self, \.$packageName)
-                    .field(Version.self, \.$spiManifest)
+                    .field(Version.self, \.$docArchives)
                     .field(Repository.self, \.$ownerName)
-                    .first(),
-                      let body = awsResponse.body,
+                    .all()
+
+                let documentationVersions = queryResult.compactMap { result -> DocumentationVersion? in
+                    guard let reference = result.model.reference else { return nil }
```

Comment:
Getting a warning here - `result.model.reference` isn't actually optional.

Reply:
This is so odd. I wasn't getting these on Xcode 14 at all. Switched back to 13 now, and they show up.